File: rdd/WordCount.scala
package com.sparkTutorial.rdd

import org.apache.log4j.Level
import org.apache.log4j.Logger
import org.apache.spark.SparkConf
import org.apache.spark._

object WordCount {

  def main(args: Array[String]) {

    Logger.getLogger("org").setLevel(Level.ERROR)
    val conf = new SparkConf().setAppName("wordCounts").setMaster("local[3]")
    val sc = new SparkContext(conf)

    val lines = sc.textFile("in/word_count.text")
    val words = lines.flatMap(line => line.split(" "))

    val wordCounts = words.countByValue()
    for ((word, count) <- wordCounts) println(word + " : " + count)
  }
}

File: rdd/WordCount.py
import sys
from pyspark import SparkContext

if __name__ == "__main__":
	sc = SparkContext("local", "word count")
  	lines = sc.textFile("in/word_count.text")
  	words = lines.flatMap(lambda line: line.split(" "))
  	wordCounts = words.countByValue()
  	for word, count in wordCounts.items():
  		print(word, count)